lib/spack/spack/extensions.py

```diff
@@ -24,39 +31,44 @@ def extension_name(path):
         path (str): path where the extension resides
 
     Returns:
-        The extension name or None if path doesn't match the format
-        for Spack's extension.
+        The extension name. An exception is raised if path doesn't match
+        the expected format for a Spack command extension.
     """
-    regexp_match = re.search(extension_regexp, os.path.basename(path))
+    regexp_match = re.search(_extension_regexp,
+                             os.path.basename(path.rstrip(os.sep)))
```

Comment:
Wondering if this might be better:
```suggestion
                             os.path.basename(os.path.normpath(path))
```

Reply:
@alalazo Are you OK with the caveat at (_e.g._) https://docs.python.org/3/library/os.path.html#os.path.normpath, specifically that, "This string manipulation may change the meaning of a path that contains symbolic links?"

Reply:
>  Are you OK with the caveat ... ?

Yes

Reply:
@alalazo maybe:
```suggestion
    os.path.basename(os.path.normpath(os.path.realpath(path)))
```  
would work for you while avoiding the possible semantic change?

Reply:
@chissg I defensively proposed `os.path.normpath` since it was not calling `stat` but was a modification on the path alone - without checking if it even exists. If `os.path.realpath` doesn't have any performance issue it would be fine too modulo deciding if resolving symlinks is something we want or not.

Reply:
After having thought about this, I am fine with `normpath()` since we're immediately calling `basename()` on the result. Committing your suggestion as-is.